kernel/mem_domain.c

```diff
@@ -22,7 +22,7 @@ static bool sane_partition(const struct k_mem_partition *part,
 	u32_t end;
 	u32_t i;
 
-	end = part->start + part->size;
+	end = part->start + part->size - 1;
```

Comment:
Drive-by naming gripe: this probably should have been called "last", not "end".  It's the address of the last byte in the region.  Usually (but not always) "end" refers to the boundary, which is going to be the address of the byte after the region (i.e. so end = start + size, which is the intuitive "fact" that got mixed up here).

Reply:
Right, so i can either
- replace "end" with "last" or "limit"
- or revert this change and modify line 42, so <, > become <=, >=
Believing that these are equivalent options, I am asking @andyross , what you would prefer :)

Reply:
It's a valid point actually, not just a style thing. Fixed